src/main/java/io/cdap/delta/bigquery/BigQueryAssessor.java

```diff
@@ -35,9 +39,25 @@
   public TableAssessment assess(StandardizedTableDetail tableDetail) {
     List<ColumnAssessment> columnAssessments = new ArrayList<>();
     for (Schema.Field field : tableDetail.getSchema().getFields()) {
-      columnAssessments.add(new ColumnAssessment(field.getName(), toBigQueryType(field).toLowerCase()));
+      try {
+        String bqType = toBigQueryType(field);
+        columnAssessments.add(ColumnAssessment.builder(field.getName(), bqType)
+                                .setSourceColumn(field.getName())
+                                .build());
+      } catch (Exception e) {
+        columnAssessments.add(ColumnAssessment.builder(field.getName(), "N/A")
+                                .setSourceColumn(field.getName())
+                                .setSupport(ColumnSupport.NO)
+                                .setSuggestion(new ColumnSuggestion(e.getMessage(), Collections.emptyList()))
```

Comment:
should we log the exception as well?

Reply:
in this particular case, no, because it will always be an exception about some unsupported type, and not some connection issue or anything where the root cause is hidden in the stacktrace. I will change it to catch IllegalArgumentException so that it's a little bit more clear